pkg/framework/plugins/removepodsviolatingnodetaints/node_taint.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package removepodsviolatingnodetaints
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+const PluginName = "RemovePodsViolatingNodeTaints"
```

Comment:
@damemi @a7i what's your opinion on preserving the strategy names as the plugin names? `RemovePodsViolatingNodeTaints` is really long. `RemovePods` part is implicit in the descheduler. What about to use `NodeTaintChecker` or similar instead?

Reply:
Agreed. In fact, `RemovePods` is misleading as it implies pod deletion, it's more delegate EvictPod to eviction API (which may or may not happen). Just `NodeTaints`?

Reply:
I had an issue open to discuss this a while back https://github.com/kubernetes-sigs/descheduler/issues/439. The redundant, long names are confusing so I think it would benefit if we can shorten them

Reply:
Summarizing:
- descheduler never removes a pod, it only evicts one => `Remove` keyword can be drop
- descheduler always evicts pods only => `Pods` keyword is redundant
- taints are always present on nodes only => should we drop `Node` keyword?

Turning the name into `ViolatingTaints`. Checking naming under https://github.com/kubernetes/kubernetes/tree/master/pkg/scheduler/framework/plugins and https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg many plugins mentions a feature. E.g. 
nodeaffinity, podtopologyspread, coscheduling, crossnodepreemption, podstate. Given we are checking against certain features, we might follow the same pattern. E.g. `Taints`, `NodeAffinity`, `TopologySpreadConstraint` (given all  plugin are evicting pods "violating" some constraint we can leave out `Violating` keyword as well). Turning all plugin names into:
- `duplicates`
- `failedpods` 
- `nodeaffinity` (https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)
- `taints` (https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
- `interpodantiaffinity` (https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity)
- `podlifetime`
- `toomanyrestarts`
- `topologyspreadconstraint` (https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/)


Plugins such as `nodeaffinity`, `taints`, `interpodantiaffinity` and `topologyspreadconstraint` operator with known features. In the remaining cases it's up to us to choose the naming.

Reply:
Personally I think `NodeTaints` is a bit more descriptive for this particular feature. But naming isn't my forte, so I'm fine with just `taints`.

I think it will be tough to establish a defined naming pattern but we can set some guidelines like this which we can point to.

Reply:
hi, very good pr, thanks @ingvagabund , may i ask is it that we need to rename the plugin name as what you mention above?
or still under discuss?

Reply:
Correct. We can continue with the discussion in https://github.com/kubernetes-sigs/descheduler/issues/439